### ISSUE TITLE
Krill Upgrade broke publisher

### ISSUE THREAD

**Issue by booth-f** [NLnetLabs/krill#965]
Hello,

So I have Krill running on a server running Ubuntu 20.04 LTS and I noticed today that the repository was giving me an error that the configured publisher CA was not configured to publish to my server. I then attempted to remove the publisher and add it back in with the existing CA Krill had already configured for me during the initial setup process.

Right now I appear to have the CA assigned back as a publisher for the server but for some reason the CA and publish URL differs so I get the following error message from both the Krill CLI and web UI: 

```
 October 21st 2022, 11:27:37 { "Failure": { "label": "rfc8181-delta", "msg": "Publishing (rsync://rpki.xxxxxxxxxx.net/repo/rpki/0/B14DDE4D1D443E8DFC063D808B410CEDF0F7AB4F.crl) outside of jail URI (rsync://rpki.xxxxxxxxxx.net/repo/RPKI/) is not allowed.", "args": { "cause": "Publishing (rsync://rpki.xxxxxxxxxx.net/repo/rpki/0/B14DDE4D1D443E8DFC063D808B410CEDF0F7AB4F.crl) outside of jail URI (rsync://rpki.xxxxxxxxxx.net/repo/RPKI/) is not allowed." } } } 
 ```

Is there a way for me to update the jail URI to permit this CA to publish into the existing repo without rebuilding the entire server?
For context this was all running fine until I recently upgraded Krill to v0.12.0-rc3

**Comment by timbru** [comment]
Hi,

That is unfortunate. Can I ask you to make a backup of your data directory and contact me on tim@nlnetlabs.nl? If you could share your data dir with me somehow then I can have a look to see why the RC upgrade failed..

To resolve your current issue..

It seems that your CA is no longer entitled to publishing in what it believes to be its 'base directory'. You probably changed the publisher name.  You can try to "migrate" the CA to a new repository and give it the updated "Repository Response" as described here: https://krill.docs.nlnetlabs.nl/en/latest/ca-migrate-repo.html

If that does not work, you may want to do a rollback - even though it's a bit risky:
https://krill.docs.nlnetlabs.nl/en/latest/install-and-run.html#rollback

**Comment by booth-f** [comment]
Hey, sorry about that. I missed the github email showing you had responded.

I ended up deleting the old config and re-initializing Krill to get it back to working. A lot had been done on that server so I'm not sure if it was Krill that broke during the update but I got it working again. I'm just using it as a small scale publication server for a single ASN so it wasn't too big of a deal to get the RPKI repo reloaded.

Thanks though for reaching out, I'll keep my eye out just in case I run into any issues again but I think it's all set now.
